src/main/java/service/impl/FileReaderImpl.java

```
@@ -0,0 +1,20 @@
+package service.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import service.FileReader;
+
+public class FileReaderImpl implements FileReader {
+
+    public List<String> read(String fileName) {
```

The method does not check for null or empty fileName parameters. According to the checklist, you should validate parameters and throw a RuntimeException if invalid (e.g., null or empty fileName).